python_ta/typecheck/base.py

```diff
@@ -473,12 +473,16 @@ def _unify_generic(self, tnode1: _TNode, tnode2: _TNode,
     def can_unify(self, t1: type, t2: type) -> bool:
         return isinstance(self.unify(t1, t2, None, False), TypeInfo)
 
+    @accept_failable
     def unify_call(self, func_type, *arg_types, node=None) -> TypeResult:
         """Unify a function call with the given function type and argument types.
 
         Return a result type.
         """
         # Check that the number of parameters matches the number of arguments.
+        if hasattr(func_type, '__args__') and isinstance(func_type.__args__[0], _ForwardRef) and isinstance(func_type.__args__[-1], _ForwardRef):
```

Comment:
Don't do this here. Instead, put logic inside `get_call_signature` to strip off the `self` argument for `__init__` specifically.